Release process document update (#283)
kyma-project/community: guidelines/releases/release-process.md

        Every component image is published with a version defined in the `RELEASE_VERSION` file stored in the `test-infra` repository on the given release branch. Test scripts for integration jobs like GKE Integration or GKE Upgrade are also loaded from the `test-infra` release branch.

>         For example for first release candidate of 1.1.0, the release version will be `1.1.0-rc1` and `yaml` files should be modified as follows:

Reviewer: ```suggestion
        For example, for the first release candidate of 1.1.0, the release version will be `1.1.0-rc1` and the `yaml` files should be modified as follows:
```

        ``` yaml
        dir:
        version: 1.1.0-rc1
        ```

>         > **CAUTION**: Do **not** update versions of the components where `dir` section does not contain `develop`. For example Console related components.

Reviewer: ```suggestion
        > **CAUTION**: Do **not** update the version of components whose `dir` section does not contain `develop`, as is the case with Console-related components.
```

7. After all checks pass, merge the PR.
    > **NOTE:** To merge the PR to the release branch, you must receive approvals from all teams.

8. Merging the PR to the release branch runs the postsubmit jobs, which:

>     - create a GitHub release and trigger documentation update on the official Kyma website.  

Reviewer: ```suggestion
    - create a GitHub release and trigger documentation update on the official Kyma website  
```

    - trigger provisioning of the cluster from the created release

>     The cluster name will contain the release version with `.` replaced with `-`. For example: `gke-release-1-1-0-rc1`. Cluster can be used for testing the release candidate.

Reviewer: ```suggestion
    The cluster name contains the release version with a period `.` replaced by a dash `-`. For example: `gke-release-1-1-0-rc1`. Use the cluster to test the release candidate.
```

>     > **CAUTION**: Release cluster needs to be deleted manually, same as other resources created by the job: IP Adresses and DNS records.

Reviewer: ```suggestion
    > **CAUTION**: The release cluster, the IP Addresses, and the DNS records must be deleted manually.
```
